src/Eshopworld.Core/IBigBrother.cs

```diff
@@ -24,6 +26,16 @@ public interface IBigBrother
         /// This should be populated by <see cref="System.Runtime.CompilerServices"/>, do not populate this manually.
         /// The line number in the source file at which the method is called.
         /// </param>
+        Task PublishAsync<T>(
+            [NotNull] T @event,
+            [CallerMemberName] string callerMemberName = "",
+            [CallerFilePath] string callerFilePath = "",
+            [CallerLineNumber] int callerLineNumber = -1) where T : TelemetryEvent;
+
+        /// <remarks>
+        /// To be removed in 4.0
+        /// </remarks>>
+        [Obsolete("Switch to PublishAsync.")]
```

Comment:
Great.

Reply:
@djfr after talking to @arturek about this. 

When this API is removed, we are left with a PublicAsync and the other Publish(Object) below. If a dev consumes this update after we remove Publish<T> method we might be left in a place where a dev pulls down the new nugets, and recompiles locally. 

At this point if they had any methods using Publish<T> they will now be using Publish(object) without really knowing so and not get the same behavior as before. For example, domain events will not be published to topics without them changing Publish<T> to PublishAsync<T> actively in all places in code.

We either we make sure they really really know the difference here and catch it in each place or we do something with Publish(object) (deprecate it for example)?